src/tiles.js

```diff
@@ -0,0 +1,152 @@
+    MAX_SPEED: { id: 6.4, traversable: false, radius: 15 },
+    SPIKE: { id: 7, traversable: false, radius: 14 },
+    BUTTON: { id: 8, traversable: false, radius: 8 },
+    INACTIVE_GATE: { id: 9, traversable: true },
+    GREEN_GATE: { id: 9.1, traversable: false },
+    RED_GATE: { id: 9.2, traversable: amRed() },
+    BLUE_GATE: { id: 9.3, traversable: amBlue() },
+    BOMB: { id: 10, traversable: false, radius: 15 },
+    INACTIVE_BOMB: { id: '10.1', traversable: true },
+    RED_TEAMTILE: { id: 11, traversable: true },
+    BLUE_TEAMTILE: { id: 12, traversable: true },
+    ACTIVE_PORTAL: { id: 13, traversable: false, radius: 15 },
+    INACTIVE_PORTAL: { id: '13.1', traversable: true },
+    SPEEDPAD_RED_ACTIVE: { id: 14, traversable: false, radius: 15 },
+    SPEEDPAD_RED_INACTIVE: { id: 14.1, traversable: true },
+    SPEEDPAD_BLUE_ACTIVE: { id: 15, traversable: false, radius: 15 },
+    SPEEDPAD_BLUE_INACTIVE: { id: 15.1, traversable: true },
+    YELLOW_FLAG: { id: 16, traversable: true },
+    YELLOW_FLAG_TAKEN: { id: '16.1', traversable: true },
+    RED_ENDZONE: { id: 17, traversable: true },
+    BLUE_ENDZONE: { id: 18, traversable: true },
+    RED_BALL: { id: 'redball', traversable: true },
+    BLUE_BALL: { id: 'blueball', traversable: true },
+  };
+  tileNames = {};
+  keys(tileInfo).forEach(key => {
+    tileNames[tileInfo[key].id] = key;
+  });
+}
+
+/*
+ * Resets stored tile info. This function exists for testing purposes, in case we want to call
+ * computeTileInfo with a different me setup for different tests.
+ */
+export function resetTileInfo() {
+  tileInfo = undefined;
+  tileNames = undefined;
+}
+
+/*
+ * @param {number} tileID - the ID of the tile whose property we want
+ * @param {String} property - the name of a property stored in tileInfo
+ * @return - the property for the input tile
+ */
+function getTileProperty(tileID, property) {
+  const tileIDString = String(tileID);
+  assert(includes(keys(tileNames), tileIDString), `Unknown tileID: ${tileID}`);
+  const tileName = tileNames[tileID];
+  return tileInfo[tileName][property];
+}
+
+/*
+ * Returns true if tileID is traversable without consequences.
+ *
+ * Traversable includes: regular floor, all flags, inactive speedpad,
+ *   inactive gate, friendly gate, inactive bomb, teamtiles, inactive
+ *   portal, endzones
+ * Nontraversable includes: empty space, walls, active speedpad, any
+ *   powerup, spike, button, enemy/green gate, bomb, active portal
+ *
+ * @param {number} tileID - the ID of the tile that should be checked for
+ * traversability
+ */
+export function isTraversable(tileID) {
+  return getTileProperty(tileID, 'traversable');
+}
+
+
+/*
+ * Is circular nontraversable object? Returns a boolean stating whether or not
+ * the given nontraversable object tile ID is a circular nontraversable object.
+ *
+ * Circular nontraversable objects include: boosts, powerups, spikes, buttons,
+ * bombs, and active portals
```

Comment:
This is my fault, but could you add the `@param {number} tileID - ` and `@return` to `isCNTO` for me?